app/reviews/create/page.tsx

```diff
@@ -1,77 +1,172 @@
+const createReviewPage: NextPage = async ({
+  searchParams
+}: {
+  searchParams?: {
+    propertyId?: string,
+    address?: string,
+  }
+}) => {
+  const propertyId = searchParams?.propertyId;
+  let address: string|undefined = undefined;
+
+  // get the address from the propertyId if it's defined
+  if (propertyId) {
+    const supabase = createClient();
+
+    const { data, error } = await supabase
+      .from("properties")
+      .select("address")
+      .eq("id", propertyId)
+      .maybeSingle();
+    
+    // Consider adding error handling here
+    // redirect to / show not-found if no property found with the provided id
+
+    address = data?.address;
+  }
+
+  if (!address) address = searchParams?.address /* For Testing, add dummy value if none provided, consider replacing with error handling as above */ || "1 Lorem Ipsum Road, Oldfield Park";
 
   // bind the id & address to the createReview function
   const initialState = { message: null, errors: {} };
-  const createReviewWithProperty = createReview.bind(null, {id: examplePropertyId, address: exampleAddress});
-  const [state, dispatch] = useFormState(createReviewWithProperty, initialState)
+  const createReviewWithProperty = createReview.bind(null, {
+    id: propertyId,
+    address: address,
+  });
+  const [state, dispatch] = useFormState(
+    createReviewWithProperty,
+    initialState
+  );
 
   return (
-    <div>
-      <h1>Create Review</h1>
-      <p>Test page for create review action - REPLACE ME</p>
+    <div style={{ minWidth: "1000px" }}>
+      <h1 className="text-4xl font-bold mb-2 mt-8">Add New Review</h1>
+      <p className="mb-8 text-xl">
+        Add a new review for this property:{" "}
+        <span>{address}</span>
+      </p>
 
       <form
-        className="flex flex-col gap-2 pleace-items-center max-w-prose w-full border p-4 rounded-md"
+        className="flex flex-col gap-2 pleace-items-center w-full border p-4 rounded-md mb-8"
         action={dispatch}
+        style={{ backgroundColor: "#778378", color: "#222" }}
       >
-        <label htmlFor="review_date">Review Date</label>
-        <input 
+        <h2 className="text-xl font-semibold">Your Details</h2>
+
+        <label htmlFor="name">Your Name</label>
+        <textarea
+          className="border p-1 rounded-md h-10"
+          name="name"
+          id="name"
+          required
+        ></textarea>
+
+        <label htmlFor="tenancy_period">Tenancy Period</label>
+        <input
           className="border p-2 rounded-md"
-          type="date"
-          name="review_date"
-          id="review_date"
+          type="text"
+          name="tenancy_period"
+          id="tenancy_period"
           required
         />
 
-        <label htmlFor="review_body">Review Body</label>
-        <textarea 
-          className="border p-2 rounded-md"
-          name="review_body"
-          id="review_body"
+        <label htmlFor="proof_of_tenancy">Upload proof of tenancy</label>
```

Comment:
Yes you're right I forgot about this. I'll fix this in the next commit. 

Should I remove the field for uploading images as well for now?

Reply:
You can probably just comment that one  out for now since we do want to add it later, but yes, I’d remove it from the initial prototype.

you could also put a placeholder there that explicitly acknowledges ‘image uploads coming soon’ or something to that effect if you want to preserve the layout